sdk/src/test/java/io/dapr/client/DaprHttpTest.java

```diff
@@ -177,16 +179,48 @@ public void invokePostMethodUnknownError() throws IOException {
 
     @Test
     public void getHeadersAndStatus(){
-
         mockInterceptor.addRule()
                 .post("http://localhost:3500/v1.0/state")
                 .respond(500, ResponseBody.create(MediaType.parse("application/json"),
                         "{\"errorCode\":\"null\",\"message\":\"null\"}"));
-
         DaprHttp daprHttp = new DaprHttp(3500, okHttpClient);
-
         System.out.println(daprHttp);
+    }
 
+    @Test ()
+    public void testCallbackCalledAtTheExpectedTimeTest() throws IOException {
+        String deletedStateKey = "deletedKey";
+        String existingState = "existingState";
+        String urlDeleteState = Constants.STATE_PATH + "/" + deletedStateKey;
+        String urlExistingState = Constants.STATE_PATH + "/" + existingState;
+        mockInterceptor.addRule()
+            .get("http://localhost:3500/" + urlDeleteState)
+            .respond(200, ResponseBody.create(MediaType.parse("application/json"),
+               deletedStateKey));
+        mockInterceptor.addRule()
+            .delete("http://localhost:3500/" + urlDeleteState)
+            .respond(204);
+        mockInterceptor.addRule()
+            .get("http://localhost:3500/" +urlExistingState)
+            .respond(200, ResponseBody.create(MediaType.parse("application/json"),
+                existingState));
+        DaprHttp daprHttp = new DaprHttp(3500, okHttpClient);
+        Mono<DaprHttp.Response> response = daprHttp.invokeAPI("GET", urlExistingState, null, null);
+        assertEquals(existingState, serializer.deserialize(response.block().getBody(), String.class));
+        Mono<DaprHttp.Response> responseDeleted = daprHttp.invokeAPI("GET", urlDeleteState, null, null);
+        Mono<DaprHttp.Response> responseDeleteKey = daprHttp.invokeAPI("DELETE", urlDeleteState, null, null);
+        assertNull(serializer.deserialize(responseDeleteKey.block().getBody(), String.class));
+        mockInterceptor.reset();
+        mockInterceptor.addRule()
+            .get("http://localhost:3500/" +urlDeleteState)
+            .respond(404, ResponseBody.create(MediaType.parse("application/json"),
+                "{\"errorCode\":\"404\",\"message\":\"State Not Fuund\"}"));
+        try {
+            responseDeleted.block();
+            fail("Expected DaprException");
```

Comment:
Does this fail without the change in DaprClientGrpcAdapter above?

Reply:
This test is completly isolated to the DaprHttp, not sure what are you asking regarding GRPC

Reply:
One of the changes in this PR is in DaprClientGrpcAdapter.java.  I thought this was validating that.  Sounds like it's not.  

Ok then is this testing the change in this PR in DaprHttp.java?  And does this test fail w/o that change?

Reply:
Test for the GRPC client has been added @LMWF can you take another look?